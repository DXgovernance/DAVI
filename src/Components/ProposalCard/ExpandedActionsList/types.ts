import { DecodedAction } from 'Components/ActionsBuilder/types';

export interface ExpandedActionsListProps {
  actions: DecodedAction[];
}
