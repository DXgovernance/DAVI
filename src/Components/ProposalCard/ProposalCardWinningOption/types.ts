import { Option } from 'Components/ActionsBuilder/types';

export interface ProposalCardWinningOptionProps {
  option: Option;
}
