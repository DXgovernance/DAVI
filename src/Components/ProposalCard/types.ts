import { ProposalStatusProps } from 'Components/ProposalStatus/types';
import { Proposal } from 'types/types.guilds.d';
import { ENSAvatar } from '../Types';
import { DecodedAction } from 'old-components/Guilds/ActionsBuilder/types';

export interface ProposalCardProps {
  proposal?: Proposal;
  votes?: number[];
  ensAvatar?: ENSAvatar;
  href?: string;
  statusProps?: ProposalStatusProps;
  summaryActions?: DecodedAction[];
}
