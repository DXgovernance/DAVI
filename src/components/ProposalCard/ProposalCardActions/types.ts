import { UseProposalVotesOfVoterReturn } from 'Modules/Guilds/Hooks/useProposalVotesOfVoter';

export interface ProposalCardActionsProps {
  votesOfVoter: UseProposalVotesOfVoterReturn;
  proposalCreator: string;
  userAddress: string;
}
