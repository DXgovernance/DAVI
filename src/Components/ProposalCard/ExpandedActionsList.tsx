import { getInfoLineView } from 'Components/ActionsBuilder/SupportedActions';
import UndecodableCallInfoLine from 'Components/ActionsBuilder/UndecodableCalls/UndecodableCallInfoLine';
import {
  ActionNumber,
  ActionsTooltipWrapper,
  ActionWrapper,
  SpacerLine,
} from './ProposalCard.styled';

const ExpandedActionsList = ({ actions }) => {
  return (
    <ActionsTooltipWrapper>
      {actions.map((action, index) => {
        const InfoLine = getInfoLineView(action?.decodedCall?.callType);

        return (
          <>
            <ActionWrapper>
              <ActionNumber>{index + 1}</ActionNumber>
              {!!InfoLine ? (
                <InfoLine
                  decodedCall={action?.decodedCall}
                  approveSpendTokens={action?.approval}
                  noAvatar
                />
              ) : (
                <UndecodableCallInfoLine />
              )}
            </ActionWrapper>
            {index !== actions.length - 1 && <SpacerLine />}
          </>
        );
      })}
    </ActionsTooltipWrapper>
  );
};

export default ExpandedActionsList;
