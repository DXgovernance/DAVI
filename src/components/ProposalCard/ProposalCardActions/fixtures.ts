import { BigNumber } from 'ethers';

export const mockProposalCardActionsProps = {
  votesOfVoter: {
    option: '1',
    votingPower: BigNumber.from(0),
  },
  proposalCreator: '0x0000000000000000000000000000000000000000',
  userAddress: '0x0000000000000000000000000000000000000000',
};
