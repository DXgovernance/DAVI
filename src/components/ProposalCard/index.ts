export { ProposalCard } from './ProposalCard';
export * from './types';
