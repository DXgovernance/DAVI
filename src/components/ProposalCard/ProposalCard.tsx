import { ProposalCardProps } from 'components/ProposalCard/types';
import { Avatar } from 'components/Avatar';
import { ProposalStatus } from 'components/ProposalStatus';
import { Loading } from 'components/primitives/Loading';
import { UnstyledLink } from 'components/primitives/Links';
import 'react-loading-skeleton/dist/skeleton.css';
import { shortenAddress } from 'utils';
import {
  CardWrapper,
  CardHeader,
  IconDetailWrapper,
  Detail,
  CardContent,
  CardTitle,
  CardFooter,
} from './ProposalCard.styled';
import ProposalCardWinningOption from './ProposalCardWinningOption/ProposalCardWinningOption';

export const ProposalCard: React.FC<ProposalCardProps> = ({
  proposal,
  ensAvatar,
  href,
  statusProps,
  options,
}) => {
  return (
    <UnstyledLink to={href || '#'} data-testid="proposal-card">
      <CardWrapper disabled={!href}>
        <CardHeader>
          <IconDetailWrapper>
            <Avatar
              src={ensAvatar?.imageUrl}
              defaultSeed={proposal?.creator}
              size={24}
            />
            <Detail>
              {ensAvatar?.ensName ||
                (proposal?.creator ? (
                  shortenAddress(proposal.creator)
                ) : (
                  <Loading style={{ margin: 0 }} loading text />
                ))}
            </Detail>
          </IconDetailWrapper>
          <ProposalStatus {...statusProps} />
        </CardHeader>
        <CardContent>
          <CardTitle size={2}>
            <strong>
              {proposal?.title || (
                <Loading style={{ margin: 0 }} loading text />
              )}
            </strong>
          </CardTitle>
        </CardContent>
        <CardFooter>
          {options && <ProposalCardWinningOption option={options[0]} />}
        </CardFooter>
      </CardWrapper>
    </UnstyledLink>
  );
};
