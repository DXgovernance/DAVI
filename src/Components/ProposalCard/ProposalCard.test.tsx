import ProposalCard from 'Components/ProposalCard/ProposalCard';
import { render } from '../../utils/tests';
import { ProposalCardProps } from 'Components/ProposalCard/types';
import {
  ensAvatarMock,
  optionsMock,
  optionsWithSeveralActionsMock,
  proposalMock,
  proposalStatusPropsMock,
} from 'Components/Fixtures';
import { BigNumber } from 'ethers';
import { mockChain } from 'Components/Web3Modals/fixtures';

jest.mock('ipfs', () => jest.fn());
jest.mock('cids', () => jest.fn());
jest.mock('axios', () => jest.fn());

const mockBigNumber = BigNumber.from(100000000);

jest.mock('hooks/Guilds/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

jest.mock('hooks/Guilds/ether-swr/erc20/useERC20Info', () => ({
  useERC20Info: () => ({
    name: 'Test ERC20',
    symbol: 'TEST',
    decimals: 18,
    totalSupply: mockBigNumber,
  }),
}));

jest.mock('wagmi', () => ({
  chain: {},
  useNetwork: () => ({ chain: mockChain, chains: [mockChain] }),
}));

const validProps: ProposalCardProps = {
  proposal: proposalMock,
  ensAvatar: ensAvatarMock,
  href: 'testUrl',
  statusProps: proposalStatusPropsMock,
  options: [optionsMock],
};

const invalidProps: ProposalCardProps = {
  proposal: null,
  ensAvatar: null,
  href: null,
  statusProps: {
    timeDetail: null,
    status: null,
    endTime: null,
  },
  options: null,
};
describe('ProposalCard', () => {
  it('ProposalCard Renders properly with data', () => {
    const { container } = render(<ProposalCard {...validProps} />);
    expect(container).toMatchSnapshot();
  });

  it('ProposalCard Renders properly with more than one option ', () => {
    const propsWithMoreThanOneOption = {
      ...validProps,
      options: [optionsMock, optionsMock],
    };

    const { container } = render(
      <ProposalCard {...propsWithMoreThanOneOption} />
    );
    expect(container).toMatchSnapshot();
  });

  it('ProposalCard Renders properly with more than one option containing more than one action', () => {
    let optionWithSeveralActions = {
      ...validProps,
      options: [optionsWithSeveralActionsMock],
    };

    const { container } = render(
      <ProposalCard {...optionWithSeveralActions} />
    );
    expect(container).toMatchSnapshot();
  });

  it('ProposalCard loading', () => {
    const { container } = render(<ProposalCard {...invalidProps} />);
    expect(container).toMatchSnapshot();
  });
});
