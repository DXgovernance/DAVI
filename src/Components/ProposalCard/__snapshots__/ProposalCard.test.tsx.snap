// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalCard ProposalCard Renders properly with data 1`] = `
<div>
   
  <a
    class="sc-dkmKIT eZXvkL"
    data-testid="proposal-card"
    href="/testUrl"
  >
    <div
      class="sc-iqHYmW sc-flMpop fkFZia fhkIyA"
    >
      <div
        class="sc-iqHYmW sc-eWVLlQ fkFZia fKXacY"
      >
        <div
          class="sc-iqHYmW sc-kUbgcP fkFZia bTUfMg"
        >
          <div
            class="paper"
            style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(245, 228, 0); height: 24px; width: 24px;"
          >
            <svg
              height="24"
              width="24"
              x="0"
              xmlns="http://www.w3.org/2000/svg"
              y="0"
            >
              <rect
                fill="#F2BE02"
                height="24"
                rx="0"
                ry="0"
                transform="translate(4.11204385070503 -3.89652025148055) rotate(387.1 12 12)"
                width="24"
                x="0"
                y="0"
              />
              <rect
                fill="#01778E"
                height="24"
                rx="0"
                ry="0"
                transform="translate(-1.9540919563657109 12.528454691642345) rotate(227.4 12 12)"
                width="24"
                x="0"
                y="0"
              />
              <rect
                fill="#C81435"
                height="24"
                rx="0"
                ry="0"
                transform="translate(13.477268130678231 -9.444486788922836) rotate(422.3 12 12)"
                width="24"
                x="0"
                y="0"
              />
              <rect
                fill="#FA9600"
                height="24"
                rx="0"
                ry="0"
                transform="translate(14.838094142329384 -25.57490104209306) rotate(308.5 12 12)"
                width="24"
                x="0"
                y="0"
              />
            </svg>
          </div>
          <div
            class="sc-iqHYmW sc-dwcupg fkFZia kWyqIM"
          >
            venky0x.eth
          </div>
        </div>
        <div
          class="sc-gsBqHK cUHEhI"
        >
          <div
            class="sc-aKXVD esWbBH"
          >
            <div
              class="sc-iqHYmW sc-cvJIAo fkFZia elhAwV"
            >
              <span
                title="May 9th, 2022 - 8:00 am"
              >
                Time
              </span>
            </div>
            <div
              class="sc-iqHYmW sc-kmATbt fkFZia jIfEqs"
            >
               
              Active
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-iqHYmW sc-iGculD fkFZia dwdZKR"
      >
        <h2
          class="sc-kstqJO sc-eWvPqa ezOHft gHjKrC"
          size="2"
        >
          <strong>
            SWPR single reward campaign
          </strong>
        </h2>
      </div>
      <div
        class="sc-iqHYmW sc-irlQje fkFZia jGFgqL"
      >
        <div
          class="sc-iqHYmW sc-jtHLbV fkFZia fyOCLY"
        />
        <div
          class="sc-iqHYmW sc-iIEXtl fkFZia hvckDF"
        >
          <div
            class="sc-iqHYmW sc-kUbgcP sc-eltbHq fkFZia bTUfMg cKtwLc"
          >
            <div
              class="sc-iqHYmW sc-dwcupg fkFZia kWyqIM"
            >
              20
              %
            </div>
            <div
              class="sc-cKZGGw GGvKv"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
              </svg>
            </div>
            <div
              class="sc-iqHYmW sc-dwcupg fkFZia kWyqIM"
            >
              10
              %
            </div>
          </div>
        </div>
      </div>
    </div>
  </a>
   
</div>
`;

exports[`ProposalCard ProposalCard loading 1`] = `
<div>
   
  <a
    class="sc-dkmKIT eZXvkL"
    data-testid="proposal-card"
    href="/"
  >
    <div
      class="sc-iqHYmW sc-flMpop fkFZia fhkIyA"
    >
      <div
        class="sc-iqHYmW sc-eWVLlQ fkFZia fKXacY"
      >
        <div
          class="sc-iqHYmW sc-kUbgcP fkFZia bTUfMg"
        >
          <div
            class="sc-bdfBQB"
            style="margin: 0px;"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="width: 24px; height: 24px; border-radius: 50%; --base-color: #333; --highlight-color: #555;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-iqHYmW sc-dwcupg fkFZia kWyqIM"
          >
            <div
              class="sc-bdfBQB"
              style="margin: 0px;"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 100px; height: 16px; border-radius: 50px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-gsBqHK cUHEhI"
        >
          <div
            class="sc-aKXVD esWbBH"
          >
            <div
              class="sc-iqHYmW sc-cvJIAo fkFZia elhAwV"
            >
              <div
                class="sc-bdfBQB"
              >
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                    style="width: 50px; --base-color: #333; --highlight-color: #555;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="sc-bdfBQB"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 50px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-iqHYmW sc-iGculD fkFZia dwdZKR"
      >
        <h2
          class="sc-kstqJO sc-eWvPqa ezOHft gHjKrC"
          size="2"
        >
          <strong>
            <div
              class="sc-bdfBQB"
              style="margin: 0px;"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 100px; height: 16px; border-radius: 50px; --base-color: #333; --highlight-color: #555;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </strong>
        </h2>
      </div>
      <div
        class="sc-iqHYmW sc-irlQje fkFZia jGFgqL"
      >
        <div
          class="sc-iqHYmW sc-jtHLbV fkFZia fyOCLY"
        />
        <div
          class="sc-bdfBQB"
          style="margin: 0px;"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 200px; --base-color: #333; --highlight-color: #555;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </div>
  </a>
   
</div>
`;
